### PR TITLE
initial build and test pipeline using actions

### PR DESCRIPTION
**Pull request by tomjebo**

### CODE REVIEW

File: .github/workflows/actions-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+name: dotnet/Open-XML-SDK main pipeline
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:1]: can probably just be "Build" or something

File: .github/workflows/actions-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+on:
+  push:
+    branches: 
+      - 'main'
+      - 'release/*'
+  pull_request:
+    branches: 
+      - 'main'
+      - 'release/*'
+      - '!/docs/'
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:16]: :) hopefully this works here

File: .github/workflows/actions-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        fetch-depth: 0
+
+    # Install the .NET Core workload
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 6.0.x
+
+    # Add  MSBuild to the PATH: https://github.com/microsoft/setup-msbuild
+    - name: Setup MSBuild.exe
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:50]: we shouldn't need this

File: .github/workflows/actions-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      uses: microsoft/setup-msbuild@v1.0.2
+
+    # Execute all unit tests in the solution
+    # - name: Execute unit tests
+    #  run: dotnet test
+
+    # Restore the projects
+    - name: Restore the Projects
+      run: msbuild $env:Solution_Name /t:Restore /p:Configuration=$env:Configuration
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:59]: we can use dotnet restore here

File: .github/workflows/actions-pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      env:
+        Configuration: ${{ matrix.configuration }}
+        ProjectLoadStyle: ${{ matrix.projectloadstyle }}
+        
+            # Restore the projects
+    - name: Build the Projects
+      run: msbuild $env:Solution_Name /p:Configuration=$env:Configuration
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:66]: dotnet build --no-restore ...

File: .github/workflows/actions-pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      env:
+        Configuration: ${{ matrix.configuration }}
+        ProjectLoadStyle: ${{ matrix.projectloadstyle }}
+        
+    - name: Run All Tests
+      run: dotnet test --no-restore --verbosity normal --configuration ${{ matrix.configuration }}
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:72]: dotnet test --no-restore --no-build

File: .github/workflows/actions-pipeline.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+        projectloadstyle: [ net35, net40, net46, net6_0, net7_0 ]
+
+    runs-on: windows-latest  # For a list of available runner types, refer to
+                             # https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+
+    env:
+      Solution_Name: Open-XML-SDK.sln                         # Replace with your solution name, i.e. MyWpfApp.sln.
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:35]: can you remove these comments that aren't really necessary once we get it set up?

**Comment by twsouthwick** [comment]
Linux/Mac test runs?

**Comment by tomjebo** [comment]
> Linux/Mac test runs?

yes, I'm adding those now.

File: .github/workflows/actions-pipeline.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Execute all unit tests in the solution
+    # - name: Execute unit tests
+    #  run: dotnet test
+
+    # Restore the projects
+    - name: Restore the Projects
+      run: msbuild $env:Solution_Name /t:Restore /p:Configuration=$env:Configuration
+      env:
+        Configuration: ${{ matrix.configuration }}
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:61]: don't need configuration here and in the line. I kin dof like using the nv block for all these projectloadstyle/configuration/etc

Comment (tomjebo) [reply .github/workflows/actions-pipeline.yml]: @twsouthwick this should be better and has the os platforms in it.

File: .github/workflows/actions-pipeline.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      - 'release/*'
+      - '!/docs/'
+      - '!/*.md'
+
+env:
+  RunPeVerify: true 
+  
+jobs: 
+
+  build:
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:24]: Can we name this with a capital?

File: .github/workflows/actions-pipeline.yml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v3
+      with:
+          dotnet-version: 6.0.x
+
+    - name: Restore the Projects
+      run: dotnet restore --verbosity normal
+      env:
+          Configuration: ${{ matrix.configuration }}
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:57]: Can these be set as env block for the whole job?

**Review by twsouthwick — APPROVED** [review]
Lgtm! A couple questions, but we can get it in

File: .github/workflows/actions-pipeline.yml [hunk 11]
```diff
@@ -0,0 +1,70 @@
+name: Build
+
+concurrency:
```
Comment (twsouthwick) [inline .github/workflows/actions-pipeline.yml:3]: do we need this?

File: .github/workflows/actions-pipeline.yml (hunk 2 above)
Comment (tomjebo) [reply .github/workflows/actions-pipeline.yml:16]: according to the docs this looks valid but I haven't tested this.

File: .github/workflows/actions-pipeline.yml (hunk 9 above)
Comment (tomjebo) [reply .github/workflows/actions-pipeline.yml:24]: Yes, how's: 

```yaml
jobs: 

  build_and_test:
    name: Build and Test
```

File: .github/workflows/actions-pipeline.yml (hunk 11 above)
Comment (tomjebo) [reply .github/workflows/actions-pipeline.yml:3]: looks like that's at the workflow level so no, I think we can nix this.

File: .github/workflows/actions-pipeline.yml (hunk 10 above)
Comment (tomjebo) [reply .github/workflows/actions-pipeline.yml:57]: yes, done.

**Review by twsouthwick — APPROVED** [review]

**Pull request merged**